### PR TITLE
Populate `contao_` Symfony translations into `$GLOBALS['TL_LANG']`

### PR DESCRIPTION
**Pull request by fritzmg**
Currently we cannot override `contao/languages/` translations via Symfony translations in `translations/`. Also we cannot provide labels for DCA fields via Symfony translations. Our goal is to move away from the old Contao translations in `$GLOBALS['TL_LANG']` and this PR provides a first step. 

This PR simply populates the `$GLOBALS['TL_LANG']` array with any Symfony translations that are found within the `contao_*` translation domains. Thus you are able to override regular Contao translations in your App like this for example:

```yaml
# translations/contao_default.en.yaml
MSC:
    goBack: Back to the Future

# equivalent to:
MSC.goBack: Back to the Future
```

And if your App or extension provides new fields for `tl_content` for example, the labels can now also be provided via Symfony translations:

```yaml
# translations/contao_tl_content.en.yaml
tl_content:
    myField:
        - Title of the field
        - Description of the field.
    anotherField:
        - Lorem ipsum title
        - Dolor sit description.
```

This makes creating translation files more comfortable when manually writing them in YAML (in case you are not using services like Transifex).

_Notes:_

* The new `LegacyGlobalsProcess` helper class extracts functionality that are both needed by the `XliffFileLoader` and `MessageCatalogue` when processing these legacy translations.
* The special handling for `tl_layout` is there due to the `reset.css` etc. option values.
* If an extension provides its labels via Symfony translations, then they can _only_ be overwritten via Symfony translations in your App as well.

### CODE REVIEW

**Comment by fritzmg** [comment]
_Another note:_ this also removes the `OutOfBoundsException` if "too many" nesting levels are used in the `id` of an XLIFF translation. I don't think this arbitrary limitation was ever necessary?

File: core-bundle/src/Translation/LegacyGlobalsProcessor.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    private static function quoteKey(string $key): int|string
+    {
+        if ('0' === $key) {
+            return 0;
+        }
+
+        if (is_numeric($key)) {
+            return (int) $key;
+        }
```
Comment (ausi) [inline core-bundle/src/Translation/LegacyGlobalsProcessor.php]:
```suggestion
        if (is_numeric($key)) {
            return (int) $key;
        }
```

seems redundant?

File: core-bundle/src/Translation/LegacyGlobalsProcessor.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+        return "'".str_replace("'", "\\'", $key)."'";
```
Comment (ausi) [inline core-bundle/src/Translation/LegacyGlobalsProcessor.php]:
```suggestion
        return var_export($key);
```

File: core-bundle/src/Translation/LegacyGlobalsProcessor.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    private static function quoteValue(string $value): string
+    {
+        $value = str_replace("\n", '\n', $value);
+
+        if (str_contains($value, '\n')) {
+            return '"'.str_replace(['$', '"'], ['\\$', '\\"'], $value).'"';
+        }
+
+        return "'".str_replace("'", "\\'", $value)."'";
```
Comment (ausi) [inline core-bundle/src/Translation/LegacyGlobalsProcessor.php]:
```suggestion
        return var_export($value);
```

File: core-bundle/src/Translation/LegacyGlobalsProcessor.php (hunk 2 above)
Comment (fritzmg) [reply core-bundle/src/Translation/LegacyGlobalsProcessor.php]: Done (though `true` was missing for the second parameter)

**Comment by fritzmg** [comment]
_Another note_: this now uses `var_export` (as discussed with @ausi) which also fixes some things that were inconsistent between `dev` and `prod`.

**Review by ausi — DISMISSED** [review]

File: core-bundle/tests/Fixtures/vendor/contao/test-bundle/Resources/contao/languages/en/default.xlf [hunk 4]
```diff
@@ -21,8 +21,10 @@
                 <target>This is the fourth target</target>
             </trans-unit>
             <trans-unit id="MSC.fifth">
-                <source>This is the\nfifth source</source>
-                <target>This is the\nfifth target</target>
+                <source>This is the
+fifth source</source>
+                <target>This is the
+fifth target</target>
```
Comment (leofeyer) [inline core-bundle/tests/Fixtures/vendor/contao/test-bundle/Resources/contao/languages/en/default.xlf:27]: If `\n` no longer works, that is a BC break, isn‘t it?

Comment (fritzmg) [reply core-bundle/tests/Fixtures/vendor/contao/test-bundle/Resources/contao/languages/en/default.xlf:27]: The handling was different between `dev` and `prod`, so it did not work properly before anyway.

Comment (leofeyer) [reply core-bundle/tests/Fixtures/vendor/contao/test-bundle/Resources/contao/languages/en/default.xlf:27]: How does the Symfony translator handle this case?

Comment (fritzmg) [reply core-bundle/tests/Fixtures/vendor/contao/test-bundle/Resources/contao/languages/en/default.xlf:27]: In YAML files you can use `|` in order to create translation labels with line breaks: https://symfony.com/doc/current/reference/formats/yaml.html#strings

Comment (leofeyer) [reply core-bundle/tests/Fixtures/vendor/contao/test-bundle/Resources/contao/languages/en/default.xlf:27]: That‘s great, but it does not answer the question. If the Symfony translator converts a literal `\n` in an .xliff file to a real line-break, the change is a BC break. However, if it is only a "feature" of our cache warmer, we can drop it.

Comment (ausi) [reply core-bundle/tests/Fixtures/vendor/contao/test-bundle/Resources/contao/languages/en/default.xlf:27]: The Symfony translator does not convert `\n` or anything else inside `<target>` (it only does a UTF-8 conversion if the .xlf file is not stored as UTF-8).

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @fritzmg.

**Pull request merged**